src/main/java/org/apache/mesos/scheduler/plan/strategy/DependencyStrategy.java

```diff
@@ -0,0 +1,24 @@
+package org.apache.mesos.scheduler.plan.strategy;
+
+import org.apache.mesos.scheduler.plan.Element;
+
+import java.util.Collection;
+
+/**
+ * A DependencyStrategy implements the strategy designated by the {@link DependencyStrategyHelper} with which it was
+ * constructed.
+ *
+ * @param <C> is the type of {@link Element}s to which the Strategy applies.
+ */
+public class DependencyStrategy<C extends Element> extends InterruptableStrategy<C> {
```

Comment:
Given how small this is, any reason to have it distinct from DependencyStrategyHelper? Feels like Helper's logic could be merged into here.